src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -138,6 +150,34 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
 
   const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
     if (!localDb) return;
+
+    queue.addPriorityJob({
+      operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveAllChat,
+      id: channel?.id,
+      priority: QueueJobPriority.HIGH,
+      task: async () => {
+        const saveChatPromises = response?.messages?.map((message) => {
+          return new Promise((resolve) => {
+            const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+            chatMessage.save(localDb);
+            resolve(true);
+          });
+        });
+
+        await Promise.all(saveChatPromises);
+      }
+    });
+
+    queue.addPriorityJob({
+      operationLabel: DatabaseOperationLabel.FetchChannelDetail_RefreshChannelList,
+      id: channel?.id,
+      priority: QueueJobPriority.MEDIUM,
+      task: async () => {
+        refresh('channelList');
+      },
+      forceAddToQueue: true
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of priority queues for database operations is a good practice for managing resources and ensuring important tasks are completed first. However, it's important to handle potential errors that might occur during these operations. Currently, if an error occurs during any of these operations, it will cause the promise to reject and may halt the execution of subsequent tasks in the queue. Consider adding error handling within each task to ensure that one failing operation doesn't stop the entire queue.
```diff
    queue.addPriorityJob({
      operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveAllChat,
      id: channel?.id,
      priority: QueueJobPriority.HIGH,
      task: async () => {
        const saveChatPromises = response?.messages?.map((message) => {
          return new Promise((resolve, reject) => {
            try {
              const chatMessage = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
              chatMessage.save(localDb);
              resolve(true);
            } catch (error) {
              console.log('Error saving chat message:', error);
              reject(error);
            }
          });
        });

        try {
          await Promise.all(saveChatPromises);
        } catch (error) {
          console.log('Error in saveChatPromises:', error);
        }
      }
    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->